src/TheThingsNetwork.cpp

```diff
@@ -312,3 +315,19 @@ void TheThingsNetwork::showStatus() {
   debugPrint(F("RX Delay 2: "));
   debugPrintLn(readValue(F("mac get rxdelay2")));
 }
+
+void TheThingsNetwork::setModemSerial(Stream& modemSerial)
+{
+  if (modemSerial.available() == 0)
+    return ;
+}
+
+void TheThingsNetwork::initFor()
```

Comment:
I don't think we really need this or `init()`. The constructor of the class could just use `Serial` and `Serial1`(and call `begin()` on them) and then if the user wants to use another Serial Port or Baud Rate he'd call `setModemSerial()`.